docs/REST API/rest-api.md

```diff
@@ -6932,12 +6936,70 @@ Deletes the session specified by ID. When the user associated with the session n
 
 ## Software
 
-- [List all software](#list-all-software)
+- [List software titles](#list-software-titles)
+- [List software versions](#list-software-versions)
 - [Count software](#count-software)
+- [Get software title](#get-software-title)
+
+### List software titles
+
+Get a list of all software grouped by software title.
+
+`GET /api/v1/fleet/software/titles`
+
+#### Parameters
+
+| Name                    | Type    | In    | Description                                                                                                                                                                |
+| ----------------------- | ------- | ----- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| page                    | integer | query | Page number of the results to fetch.                                                                                                                                       |
+| per_page                | integer | query | Results per page.                                                                                                                                                          |
+| order_key               | string  | query | What to order results by. Allowed fields are `name`, `hosts_count`, `vulnerabilities_count`. Default is `hosts_count` (descending).      |
+| order_direction         | string  | query | **Requires `order_key`**. The direction of the order given the order key. Options include `asc` and `desc`. Default is `asc`.                                              |
+| query                   | string  | query | Search query keywords. Searchable fields include `title` and `cve`.                                                                                             |
+| team_id                 | integer | query | _Available in Fleet Premium_ Filters the software to only include the software installed on the hosts that are assigned to the specified team.                             |
+| vulnerable              | bool    | query | If true or 1, only list software that has detected vulnerabilities. Default is `false`.                                                                                    |
+
+#### Example
+
+`GET /api/v1/fleet/software/titles`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "counts_updated_at": "2022-01-01 12:32:00",
+  "software_titles": [
+    {
+      "id": 12,
+      "name": "Firefox.app",
+      "versions_count": 3,
+      "source": "apps",
+      "hosts_count": "48",
+      "vulnerabilities_count": 3
```

Comment:
```suggestion
```

I think we can cut `vulnerabilities_count`. Why? The user can get this count by counting the number of items in the `vulnerabilities` array. 

It looks like we don't have a `vulnerabilities_count` in the existing `GET /software` endpoint: https://fleetdm.com/docs/rest-api/rest-api#list-all-software

What do you think?

Reply:
It will require frontend to do calculations for each `versions` array item, but should be ok.